Admin/AbstractAdmin.php

```diff
@@ -1626,6 +1626,10 @@ public function hasParentFieldDescription()
      */
     public function setSubject($subject)
     {
+        if (get_class($subject) != $this->class) {
```

Comment:
Not sure this will work well with the single inheritance functionality of Sonata. Maybe try this instead : `!is_a($subject, $this->class, true)`